### DISCUSSION TITLE
Pocketbase as a full-stack web server?

### DISCUSSION THREAD

**Discussion by lorikku** [pocketbase/pocketbase#2063] [Q&A] [answered]
Hi! So far Pocketbase looks very nice, and I just had one quick question.

I noticed that a website can be served from `/pb_public`. Is it safe to assume that a full-on front-end using eg. React can be served from here, or should I use a separate web server to serve my front-end, and keep pocketbase for back-end strictly?

I guess my question is: is Pocketbase full-stack-compatible, or should it really only be used for back-end purposes?

Thanks a lot in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
Only static files can be served from the `/pb_public` directory (eg. html, images, js, css, etc.).

> Is it safe to assume that a full-on front-end using eg. React can be served from here

Yes, as long as you are not using SSR and the entire frontend app is client-side (eg. SPA). The `--indexFallback` flag is enabled by default, so it should work even if your frontend app uses pretty urls.

> is Pocketbase full-stack-compatible, or should it really only be used for back-end purposes?

I'm not sure that I understand what do you mean by "_full-stack-compatible_".

The prebuilt executable allows to serve static files and comes with REST-ish API for common operations over your data.

When [using as framework](https://pocketbase.io/docs/use-as-framework/) you can do almost whatever you want, including defining custom routes, console commands, execute complex custom queries, etc.
